### PR TITLE
[WIP] Core/Packets: implement ScenarioPackets

### PR DESCRIPTION
**Pull request by funjoker**
**Changes proposed:**
-  added "correct" SMSG_SCENARIO_STATE structure

**Target branch(es):** 6.x

**Issues addressed:** 

**Tests performed:** (Does it build, tested in-game, etc)

**Known issues and TODO list:**
- [ ] Scenariosystem implementation
- [ ] Reason for UnkLegion
- [ ] Time might not be correct

### CODE REVIEW

File: src/server/game/Server/Packets/ScenarioPackets.cpp [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    for (uint32 i = 0; i < StepProgressIndex.size(); ++i)
+    {
+        _worldPacket << uint32(StepProgressIndex[i].Id);
+    }
+
+    _worldPacket << uint8(ScenarioCompleted);
+
+    for (uint32 i = 0; i < CriteriaProgress.size(); ++i)
+    {
+        _worldPacket << uint32(CriteriaProgress[i].Id);
```
Comment (Shauren) [inline src/server/game/Server/Packets/ScenarioPackets.cpp]: this is very bad, this loop is shared with AllAchievementData - it needs to be refactored to expose
operator<<(WorldPackets::Achievement::CriteriaProgress const&) - and you should include it here

File: src/server/game/Server/Packets/ScenarioPackets.h [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+            WorldPacket const* Write() override;
+
+            CriteriaProgress criteriaProgress;
+        };
+
+        class ScenarioCompleted final : public WorldPackets::ServerPacket
+        {
+        public:
+            ScenarioCompleted(uint32 scenarioId) : ScenarioId(scenarioId), ServerPacket(SMSG_SCENARIO_COMPLETED, 4) { }
```
Comment (tkrokli) [inline src/server/game/Server/Packets/ScenarioPackets.h]:
```
In file included from /home/travis/build/TrinityCore/TrinityCore/src/server/game/Server/Packets/ScenarioPackets.cpp:18:
/home/travis/build/TrinityCore/TrinityCore/src/server/game/Server/Packets/ScenarioPackets.h:90:52: fatal error: 
      field 'ScenarioId' will be initialized after base
      'WorldPackets::ServerPacket' [-Wreorder]
            ScenarioCompleted(uint32 scenarioId) : ScenarioId(scenarioId...
                                                   ^
```

File: src/server/game/Server/Packets/ScenarioPackets.cpp (hunk 1 above)
Comment (funjoker) [inline src/server/game/Server/Packets/ScenarioPackets.cpp]: Ok will Check it After work

**Comment by Aokromes** [comment]
In file included from /home/travis/build/TrinityCore/TrinityCore/src/server/game/Server/Packets/ScenarioPackets.cpp:18:

/home/travis/build/TrinityCore/TrinityCore/src/server/game/Server/Packets/ScenarioPackets.h:90:52: fatal error: 

      field 'ScenarioId' will be initialized after base

      'WorldPackets::ServerPacket' [-Wreorder]

            ScenarioCompleted(uint32 scenarioId) : ScenarioId(scenarioId...

**Comment by tkrokli** [comment]
Yes, the same message as I referred to in https://github.com/TrinityCore/TrinityCore/pull/17790#discussion_r74890615 .

**Comment by funjoker** [comment]
Someone else need to focus. No time atm

**Pull request closed without merging**